### ISSUE TITLE
schemaTypeCheck ignores properties under allOf when checking for existence of required fields

### ISSUE THREAD

**Issue by LittleGreenYoda42** [daveshanley/vacuum#510]
Hey there,
when the schemaTypeCheck checks for the existence of properties which are required, it does not take into account the information in an allOf section. (see example below)

The schema is actually valid but the result show the error message: `object contains required fields but no properties`

And as a small side note: The [schemaTypeCheck page](https://quobix.com/vacuum/functions/openapi/schematypecheck/) references the wrong Function source ([duplicated_enum_entry](https://github.com/daveshanley/vacuum/blob/main/functions/openapi/duplicated_enum_entry.go) instead of [schema_type](https://github.com/daveshanley/vacuum/blob/main/functions/openapi/schema_type.go))

```
openapi: 3.0.1
info:
  title: Minimal OpenAPI Example
  version: 1.0.0
paths:
  /:
    get:
      responses:
        '200':
          description: A simple get endpoint

components:
  schemas:
    BaseSchema:
      type: object
      properties:
        id:
          type: integer
          format: int64
          example: 1
        name:
          type: string
          example: "Base Name"
      
    Example:
      type: object
      required:
        - extraField
        - id
      allOf:
        - $ref: '#/components/schemas/BaseSchema'
        - type: object
          properties:
            extraField:
              type: string
              example: "Additional Data"
```

**Comment by daveshanley** [comment]
will address soon.

**Comment by VictorPascualV** [comment]
Hey @daveshanley, any updates in this one? Our team is blocked by that, and we'd like to know if a fix will land soon 🙏

**Comment by daveshanley** [comment]
I am working on a release that contains other fixes and updates. It should be ready soon, however I am tackling multiple dimensions across multiple projects at the moment. Once the stack is stable I will release it.

**Comment by daveshanley** [comment]
This check has now been added. Fixed in `v0.11.0`

https://github.com/daveshanley/vacuum/blob/main/functions/openapi/schema_type.go#L266

Multiple tests added: 

https://github.com/daveshanley/vacuum/blob/main/functions/openapi/schema_type_test.go#L1077
https://github.com/daveshanley/vacuum/blob/main/functions/openapi/schema_type_test.go#L1036
https://github.com/daveshanley/vacuum/blob/main/functions/openapi/schema_type_test.go#L998

And more.
